### PR TITLE
Add support for automatic translations

### PR DESCRIPTION
**Pull request by nuno-vieira**
### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/602

### 🎯 Goal
Add support for automatic translation when receiving messages; the user has the language set when connecting a user.

### 📝 Summary
- Allow providing a language when connecting a user
- Allow enabling automatic translation through `Components.default.messageAutoTranslationEnabled`
- Parse the language information for each user
- Parse the original language from each message

### 🛠 Implementation
A new option, `.translation`, was added to `ChatMessageLayoutOption`, which is populated if the automatic translation is enabled and if a translation exists and is valid.

### 🧪 Manual Testing Notes
Note: Automatic translation is enabled by default on the DemoApp

1. Open the DemoApp
2. Go to configuration in the initial screen
3. Set a language for the user
4. Open a Channel which both users have their language set (ex 1-1 Luke vs Han Solo)
5. Send a message from the participant device
6. A "translated to" label should appear
7. If the message is not translatable, it should not appear that it was translated

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should be manually QAed
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

**Review by martinmitrevski — COMMENTED** [review]
Code-wise looks great 👌  It's important that we add docs - it's not clear if the `translationLabel` is shown by default if the layout options are setup, or do I need to do some additional setup.
Snapshot tests can go in a separate PR as well, after the release.

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift [hunk 1]
```diff
@@ -366,29 +370,30 @@ open class ChatMessageContentView: _View, ThemeProvider, UITextViewDelegate {
         }
 
         // Metadata
-        if options.hasFootnoteOptions {
```
Comment (polqf) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift:369]: This is not checked anymore?

Comment (nuno-vieira) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift:369]: This is not needed. We now check if we have footnote views, if yes, we render the footnote. The reason is that, the translation label is shown, independent of the lastInSequence logic

**Comment by nuno-vieira** [comment]
> Code-wise looks great 👌 It's important that we add docs - it's not clear if the `translationLabel` is shown by default if the layout options are setup, or do I need to do some additional setup. Snapshot tests can go in a separate PR as well, after the release.

I've already added the snapshot tests. They were important since they helped me find 2 issues 👍

**Review by polqf — APPROVED** [review]

**Review by martinmitrevski — APPROVED** [review]

**Review by martinmitrevski — APPROVED** [review]

**Review by polqf — APPROVED** [review]

File: Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift [hunk 2]
```diff
@@ -225,30 +234,31 @@ open class QuotedChatMessageView: _View, ThemeProvider, SwiftUIRepresentable {
     /// the attachments preview of the message, or if you want to support your custom attachment.
     /// - Parameter message: The message that contains all the attachments.
     open func setAttachmentPreview(for message: ChatMessage) {
+        let currentMessageText = textView.text
         if let filePayload = message.fileAttachments.first?.payload {
             attachmentPreviewView.contentMode = .scaleAspectFit
             attachmentPreviewView.image = appearance.images.fileIcons[filePayload.file.type] ?? appearance.images.fileFallback
-            textView.text = message.text.isEmpty ? filePayload.title : message.text
+            textView.text = message.text.isEmpty ? filePayload.title : currentMessageText
```
Comment (martinmitrevski) [inline Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:241]: Hm, shouldn't we check whether `textView.text.isEmpty` instead? This feels a bit strange.

File: Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift [hunk 3]
```diff
@@ -225,30 +234,31 @@ open class QuotedChatMessageView: _View, ThemeProvider, SwiftUIRepresentable {
...
         } else if let imagePayload = message.imageAttachments.first?.payload {
             attachmentPreviewView.contentMode = .scaleAspectFill
             setAttachmentPreviewImage(url: imagePayload.imageURL)
-            textView.text = message.text.isEmpty ? "Photo" : message.text
+            textView.text = message.text.isEmpty ? "Photo" : currentMessageText
```
Comment (martinmitrevski) [inline Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:245]: Didn't we localize these at some point? Or was that a different part?

**Review by martinmitrevski — COMMENTED** [review]
2 questions before merging

File: Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift (hunk 3 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:245]: This is a different part, this one is not localized yet. Currently it is out of scope of this PR

File: Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift (hunk 2 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift:241]: Yes that could be a possiblity as well

Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/QuotedChatMessageView/QuotedChatMessageView.swift]: @martinmitrevski I made this more clear now

**Review by testableapple — APPROVED** [review]
LGTM 👍

**Pull request merged**